libs/src/api/track.js

```diff
@@ -124,6 +125,91 @@ class Tracks extends Base {
     return trackId
   }
 
+  /**
+   * Takes in a readable stream if isServer is true, or a file reference if isServer is
+   * false.
+   * WARNING: Uploads file to creator node, but does not call contracts
+   * Please pair this with the addTracksToChainAndCnode
+   */
+  async uploadTrackContentToCreatorNode (
+    trackFile,
+    coverArtFile,
+    metadata,
+    onProgress
+  ) {
+    this.REQUIRES(Services.CREATOR_NODE)
+    this.FILE_IS_VALID(trackFile)
+
+    if (coverArtFile) this.FILE_IS_VALID(coverArtFile)
+
+    this.IS_OBJECT(metadata)
+
+    const owner = this.userStateManager.getCurrentUser()
+    if (!owner.user_id) {
+      throw new Error('No users loaded for this wallet')
+    }
+
+    metadata.owner_id = owner.user_id
+    this._validateTrackMetadata(metadata)
+
+    // Upload metadata
+    const { metadataMultihash, metadataFileUUID } = await retry(async (bail, num) => {
+      return this.creatorNode.uploadTrackContent(
+        trackFile,
+        coverArtFile,
+        metadata,
+        onProgress)
+    }, {
+    // Retry function 3x
+    // 1st retry delay = 500ms, 2nd = 1500ms, 3rd...nth retry = 4000 ms (capped)
+      minTimeout: 500,
+      maxTimeout: 4000,
+      factor: 3,
+      retries: 3,
+      onRetry: (err, i) => {
+        if (err) {
+          // eslint-disable-next-line no-console
+          console.log('Retry error : ', err)
+        }
+      }
+    })
+    console.error('New upload func')
+    return { metadataMultihash, metadataFileUUID }
+  }
+
+  /**
+   * Takes an array of [{metadataMultihash, metadataFileUUID}, {}, ]
+   * Adds tracks to chain for this user
+   * Associates tracks with user on creatorNode
+   */
+  async addTracksToChainAndCnode (trackMultihashAndUUIDList) {
+    this.REQUIRES(Services.CREATOR_NODE)
+    const owner = this.userStateManager.getCurrentUser()
+    if (!owner.user_id) {
+      throw new Error('No users loaded for this wallet')
+    }
+
+    let trackIds = (await Promise.all(
+      trackMultihashAndUUIDList.map(async trackInfo => {
```

Comment:
I don't think we want the actual function passed to map to be async. I'm not even sure exactly what behavior this results in... but in general the pattern we want here is to _synchronously_ map through some elements and return an array of unresolved promises, which you call Promise.all on. In this code, the actual mapping function is async. We can either wrap the mapped element in a new Promise, or return a called async function. Example:

https://flaviocopes.com/javascript-async-await-array-map/